projects/i18n/schematics/ng-generate/lib-resources-module/lib-resources-module.schematic.ts

```diff
@@ -0,0 +1,171 @@
+import { normalize, strings } from '@angular-devkit/core';
+import { ProjectDefinition } from '@angular-devkit/core/src/workspace';
+import {
+  apply,
+  applyTemplates,
+  chain,
+  forEach,
+  MergeStrategy,
+  mergeWith,
+  move,
+  Rule,
+  Tree,
+  url,
+} from '@angular-devkit/schematics';
+
+import path from 'path';
+
+import { readRequiredFile } from '../../utility/tree';
+import { getProject, getWorkspace } from '../../utility/workspace';
+
+import { Schema } from './schema';
+import { TemplateContext } from './template-context';
+
+/**
+ * Standardize keys to be uppercase, due to some language limitations
+ * with lowercase characters.
+ * See: https://stackoverflow.com/questions/234591/upper-vs-lower-case
+ */
+function parseLocaleIdFromFileName(fileName: string): string {
+  return fileName
+    .split('.json')[0]
+    .split('resources_')[1]
+    .toLocaleUpperCase()
+    .replace('_', '-');
+}
+
+function getResources(tree: Tree, project: ProjectDefinition): string {
+  let imports: string = '';
+  let resourcesVar: string =
+    'const RESOURCES: { [locale: string]: SkyLibResources } = {';
+
+  const localesPath = normalize(`${project.sourceRoot}/assets/locales`);
+  const localesDir = tree.getDir(localesPath);
+
+  localesDir.subfiles.forEach((file) => {
+    const localeId = parseLocaleIdFromFileName(file);
+    const variableName = localeId.replace('-', '_').toLowerCase();
+    imports += `import ${variableName} from '${localesPath}/${file}';\n`;
+    resourcesVar += `\n  '${localeId}': ${variableName},`;
+  });
+
+  resourcesVar += '\n};';
+
+  return imports + '\n' + resourcesVar;
+}
+
+/**
+ * Adds `@skyux/i18n` to the project's package.json `peerDependencies`.
+ */
+function addI18nPeerDependency(project: ProjectDefinition): Rule {
```

Comment:
Why is `@skyux/assets` a hard dependency while `@skyuxi18n` is a peer dependency?

Reply:
The `ng-add` schematic is installing both `@skyux/i18n` and `@skyux/assets` on the consuming _workspace's_ dependencies (this doesn't necessarily affect the library's source code). The `ng generate` schematic adds `@skyux/i18n` to the _library's_ peer dependencies because the act of generating the resources module adds a peer requirement on `@skyux/i18n`.